### PR TITLE
feat: a lone capital after a pause is read as a sentence boundary

### LINKED ISSUE #273: A capital with no mark in front of it is never read as a sentence boundary
A pause mid-sentence sometimes makes the recogniser write a capital with no mark in front of it: `imports name definitions And all the things`. `SentenceJoin.boundaries()` only starts a boundary at a period or a question mark, so this shape is never scored and the capital ships. In one speaker's corpus it was 35 of 790 split boundaries (#256), and it hit twice in tonight's dictation. Every capital that is not a name should be read by the same three readings a period boundary gets, and lowercased where the join wins.

<details>
<summary>The 0.750 AUC in #256 was measured with the probe that has since been replaced</summary>

#256 scored the bare capital with the old threshold probe: `log P(".") - log P(" next word")` on ModernBERT, the design that repaired 26% of period cuts. On that probe the free slot after `...and the word app` was happy to take a period, and the live example scored +3.52 toward a break. The shipped design (#260, #267) scores three full readings of the continuation per token and takes the winner. It took periods from 26% to 82%. It has not been run on the bare-capital shape.

</details>

<details>
<summary>Why the capital is left alone today</summary>

`SentenceJoin.swift:109` `boundaries()` walks the text from mark to mark. A capital with nothing in front of it is not visited. docs/transcription.md:514 records this as a measured decision, on the old probe.

The cases that must not be touched: a name mid-sentence (`on Vercel`, `the Databricks notebook`), `I`, a word in capitals (`API`), a vocabulary term. `SentenceJoin.written()` already knows those rules for the word after a joined period; the same rules can filter candidates before they are read.

</details>

<details>
<summary>What the fix would touch</summary>

- `SentenceJoin.boundaries()`: a second candidate shape, a capitalised word with no mark before it, filtered by the `written()` rules before any reading.
- `SentenceReadings.build()`: the readings for a boundary that holds no mark. The join reading is the text as decoded.
- A labelled bench of bare capitals mined from the trace corpus, three piles: join (spurious capital), break (real sentence start, period dropped), other (restart). The constraint is zero joins on the break pile, as for #256.
- Whether to insert the period where the break reading wins is a separate decision. The first change is lowercase-on-join only.

Not in scope: the capital at the start of a clip that continues a typed sentence (`such that | The boilerplate`). That needs the captured input field and is a different mechanism.

</details>

### PR DESCRIPTION
**Pull request by znat**
A pause does not always make the transcriber write the mark. It writes `imports name definitions And all the things`, and `SentenceJoin.boundaries()` only ever started a boundary at a period or a question mark, so that capital shipped. It is scanned now.

The shape needs a **fourth reading**: the text exactly as it was decoded. Everywhere else "leave it alone" is what happens when a mark wins; here there is no mark to take out, so it has to be a candidate of its own. Half of these capitals are correct — `Slack`, `English`, `Friday`, `TypeScript` — so four rules refuse a candidate before any reading, and the fourth reading catches what they miss.

Measured over 7659 English dictations: **3.1 candidates per 100 dictations reach the model, 2.7 after the pause gate. On 110 hand-labelled candidates, 64% of the spurious capitals are lowercased, no correct capital is touched, and none of 15 real sentence starts is joined.** The period is never inserted; where a mark wins the text is left as decoded and the decision is logged. The period and question shapes are byte-identical — `written()` is untouched.

Reviewers: start at `SentenceJoin.bareBoundaries` and `SentenceJoin.readable`. The risky part is the rebuild when there is no mark to remove.

Closes #273.

<details>
<summary>The four readings, and why the fourth is the one that matters</summary>

```
". A before section B"     the sentence really ended
" A before section B"      as decoded, a capital that belongs
" a before section B"      a pause cut one sentence in two
", a before section B"     it is really a comma
```

Argmax per token, as today, and `join` is last so an exact tie leaves the text alone. Only the third reading writes anything.

One ender, not every configured mark. Which sentence mark belongs here is a question this stage does not answer, so a second ender buys a fifth forward pass and no decision. Scored with `?` in the set as well, it won 12 of 621 boundaries and changed nothing, because an ender win means "leave as decoded" either way.

The fourth reading is worth 4 wrong writes and costs nothing:

| readings | filter | wrong writes | repaired |
|---|---|---:|---:|
| three | none | 28 | 46/75 (61%) |
| **four** | none | **9** | 46/75 (61%) |
| three | the rules below | 4 | 44/69 (64%) |
| **four** | **the rules below** | **0** | **44/69 (64%)** |

It took 19 cases off `join` without the filter and 4 with it. **Every one of them is a capital that was correct, and not one repair was lost.** The four it saves:

```
A         as-decoded -3.461 > join -3.639   ...what's in common between section A...
A         as-decoded -3.348 > join -3.781   ...can the stage return a list of terms...
A         as-decoded -3.790 > join -4.181   ...the difference between A and B...
Outlook   as-decoded -4.376 > join -4.565   ...difficult to paste into Outlook...
```

</details>

<details>
<summary>Four rules refuse a candidate before any reading — half of them are names</summary>

`written()` decides how to write the word after a mark that is already there, where a capital that is right is rare. Here it is half the input. Over 621 candidates that survive `written()`, hand-labelled in a random sample of 120: **50% are a capital that must not be touched**, 32% a spurious capital, 7% a real sentence start, 12% a stutter.

On top of the existing rules (`I`, capitals throughout, a vocabulary term, a name tag, no lemma):

| rule | catches | why |
|---|---|---|
| a run of two or more capitals | `Better Stack`, `Hugging Face`, `Red Crawl` | 53% names and 13% boundaries — a different question |
| a capital inside the word | `WhatsApp`, `TypeScript`, `OpenAI` | `written()` only refuses capitals *throughout* |
| the part of speech | `Slack`, `English`, `Friday` | a noun or adjective here is a name, a product or a title |
| a pause under a second | — | latency, see below |

The part of speech comes from `.nameTypeOrLexicalClass`, **the tag `written()` already asks for**. Over the same 621 it gives the same class as `.lexicalClass` on 618 and the same filter decision on 620, so this is one `guard` on a value the tagger already produced, not a second pass. Asking for both schemes on one tagger changes the answer on 3 of the 621, so `readable()` builds the tagger the way `written()` builds it.

The class filter alone removes 93% of the capitals that must not be touched, and costs 9 repairs of which only 4 were reachable at all.

</details>

<details>
<summary>The pause gate is for latency, not safety — 13% fewer Qwen calls</summary>

`apply(to:config:words:)` takes the decoder's words and skips a bare capital with less than 1.0 s of silence in front of it.

| | per 100 dictations |
|---|---:|
| survive `written()` today | 8.1 |
| pass this PR's candidate shape | **3.1** |
| and the 1.0 s gate | **2.7** |

**13% of the candidates never reach the model.** At 52 ms a reading that is 0.21 ms per dictation on average, and a whole 52 ms saved on the 13% of dictations that would have paid it.

It is not the safety mechanism, and the PR body should not be read that way: with the filter and the four readings it is **0 wrong writes with the gate and 0 without**, and 44/69 repaired either way. It also *shrinks* the evidence — it removes 5 of the 19 labelled break+keep cases, taking the 95% bound from 0.08 to 0.09 wrong writes per 100 dictations. Its whole value is the forward passes it does not spend.

The alignment is safe. `Transcriber.swift:479` is `var text = result.text` and its comment says why that point was chosen: "the last point where the words still line up with the audio they came from." Nothing rewrites `text` between there and the join at `:520`. All three decode paths keep text and timings together — the first pass, the long-pause retry (`restoreTimings` maps every timestamp back to the original timeline) and `audio.second_opinion` (`unpadTimings` takes the pad off). `pauses(in:words:)` rebuilds the text from the words and returns no gate at all if they do not match, so a caller with words from somewhere else gets every candidate read rather than the wrong one skipped.

`--sentence-join` and `--sentence-probe` pass no words and apply no gate, so the harness and the tests see every candidate.

</details>

<details>
<summary>What was measured — 7659 dictations, 200 hand labels</summary>

Corpus: 31525 trace records, deduped by `wav`, English only, mined from `asr.text` — the raw decode this stage runs on, not `final`.

| | n | per 100 dictations |
|---|---:|---:|
| English dictations | 7659 | |
| bare capitals, 4 words each side | 2433 | 31.8 |
| refused by `written()` | 1645 | 21.5 |
| survive it | 621 | 8.1 |
| **pass this PR's candidate shape** | **236** | **3.1** |
| and the pause gate | 205 | 2.7 |

200 candidates hand-labelled into four piles — 120 drawn at random from the 621 and 80 more from what the class filter passes. With the filter and the four readings, on the 110 that are this PR's shape:

| pile | n | join | as-decoded | `,` | `.` |
|---|---:|---:|---:|---:|---:|
| join — spurious capital | 69 | **44 (64%)** | 0 | 23 | 2 |
| break — real sentence start | 15 | **0** | 0 | 11 | 4 |
| keep — the capital is correct | 4 | **0** | 4 | 0 | 0 |
| other — stutter or restart | 22 | 4 | 0 | 14 | 4 |

**0 wrong writes. 95% upper bound 0.08 per 100 dictations, against 1.2 repairs.**

Latency: 52.3 ms median per boundary, p95 68.5 ms, against 52.2 / 60.3 for three readings. The fourth reading goes into the same padded batch.

</details>

<details>
<summary>What was tried and left out</summary>

**A one-letter rule** (skip `section A`, `A dot B`). It caught 3 wrong writes and cost 6 repairs. The fourth reading handles all three of those cases and costs nothing, so the rule is not here.

**The comma as a repair.** On this shape `,` also lowercases, so "lowercase unless `.` wins" is a real option: it repairs 92% instead of 64%. It is out because the break pile never picks `join` but picks `,` 11 times of 15 — it would lowercase 11 real sentence starts.

**Inserting the period** where the `.` reading wins. 4 of the 15 break cases would have gained a correct period. That is a separate decision and this PR never inserts a mark.

**A world-knowledge arbiter for the run of capitals** — score the rest of the sentence with the capitals against without them, argmax, no threshold. It reads `Better Stack` the monitoring product correctly and lowercases `the Cloed Design Project`. It scored 16 of 16 on the decisive labelled runs, but two of its three original errors were relabelled after it was scored, so that number is not earned. The run shape is left for its own issue.

**Widening the vocabulary guard** to match a term with its spaces removed. `BetterStack`, `RedCrawl` and `RedRock` are one word in the vocabulary and two in the decode, and `written()` compares one word, so it misses them — 8 of 82 runs. Left out deliberately: `Better Stack` is genuinely ambiguous between the product and "a better stack", and no spelling rule settles it. `written()`'s handling of the word after a joined period is unchanged.

**Not measured.** The bench was mined with at least 4 words either side, so a bare capital with less context than that is unmeasured. The existing period shape has the same property. Also out of scope: a capital that starts a clip and continues a sentence already typed (`such that | The boilerplate`) — that needs the captured input field.

</details>

<details>
<summary>Review notes — where the risk is</summary>

**`Boundary.mark` is now optional.** nil means a bare capital, and `at` is then the first letter of the word rather than a character to remove. Two places branch on it:

- `joining()` returns the text unchanged and the word's own offset.
- the rebuild in `apply()` steps over `at` only when there is a mark there.

**The order of the scan.** `bareBoundaries()` is merged with `boundaries()` and sorted by `next.lowerBound`, so boundaries are still visited in text order and `cursor` never goes backwards.

**No double-visiting.** A bare candidate requires the previous word to end in a letter or a digit, so a capital that already follows a mark, a bracket or a quote belongs to the existing shape only.

**The filter runs before the read, never after** — a name must not reach a reading that could vote to lowercase it. `written()` moved above the `read()` call for that. It is pure, so the period path is unaffected.

**`--sentence-probe --bare`** reads a boundary as this shape from the command line, so the four readings are debuggable without a bench file. `"mark": ""` in a bench row is the same thing. `docs/cli.md` has both.

**Tests.** `tests/sentence-case-cases.json` gains 8 model-free cases for the new refusals and runs in CI. `tests/sentence-boundary-cases.json` gains 5 bare boundaries with their four stored readings; `check-sentence-probe.sh` and `check-sentence-join.sh` both read them. `check-sentence-case.sh` now picks the line for the case's own boundary rather than the first printed — `what I meant. Which one` holds a bare `I` before the period.

`make test`: every check passes, and the period and question rows are unchanged at 0 false joins.

</details>

### CODE REVIEW

File: Sources/ParrotFlow/main.swift [hunk 1]
```diff
@@ -165,13 +165,18 @@ if let at = arguments.firstIndex(of: "--sentence-probe") {
             cases: arguments[bench + 1], out: out, vectors: arguments.contains("--vectors")
         ))
     }
-    guard arguments.indices.contains(at + 2) else {
-        print("usage: ParrotFlow --sentence-probe \"<left half>\" \"<right half>\"")
+    let halves = arguments[(at + 1)...].filter { !$0.hasPrefix("--") }
```
[the remark this answers is not in the document]
Comment (znat) [reply Sources/ParrotFlow/main.swift]: Fixed in eeabe4c. Only `--bare` is taken as an option now, and only the first occurrence; everything else is transcript, taken by position.

Verified against your exact case:

```
$ ParrotFlow --sentence-probe --bare "--help" "And continue with the rest of it"
  prefix    --help
  reading   .           -4.1929  8
  reading   as-decoded  -4.6444  7
  reading   ,           -3.5767  8
  reading   join        -4.0855  7
  winner    ,
```

The usage branch still fires when only one half is given.

Note the same pattern is in the `--sentence-join` dispatcher a few lines below (`arguments[(at + 1)...].first { !$0.hasPrefix("--") }`). That one predates this PR, so I have left it alone rather than widen the diff.
